### ISSUE TITLE
QwQ-32B-Q5_K_M.gguf使用xllamacpp重复答案

### ISSUE THREAD

**Issue by bleakie** [xorbitsai/inference#3093]
### System Info / 系統信息

使用v1.3.1docker，启动后设置Temperature=0.6 and TopP=0.95无效，提示无这些参数，测试问答，max-token=2046，后面一直重复

### Running Xinference with Docker? / 是否使用 Docker 运行 Xinfernece？

- [x] docker / docker
- [ ] pip install / 通过 pip install 安装
- [ ] installation from source / 从源码安装

### Version info / 版本信息

v1.3.1

### The command used to start Xinference / 用以启动 xinference 的命令

USE_XLLAMACPP=1 xinference-local

### Reproduction / 复现过程

1.启动USE_XLLAMACPP=1 xinference-local
2.加载QwQ-32B-Q5_K_M.gguf

### Expected behavior / 期待表现

解决参数可配置，答案重复输出

**Comment by qinxuye** [comment]
@codingl2k1 xllamacpp 有正确支持这些参数么？

**Comment by bleakie** [comment]
请问有测试重复回答的问题吗？

**Comment by qinxuye** [comment]
@codingl2k1 无法重现。参数应该别正确传递到了 xllamacpp。

**Comment by bleakie** [comment]
2025-04-10 08:41:10,151 xinference.model.llm.llama_cpp.core 367 ERROR    Failed to set the param top_p = 0.95, error: 'xllamacpp.xllamacpp.CommonParams' object has no attribute 'top_p'
2025-04-10 08:41:10,151 xinference.model.llm.llama_cpp.core 367 ERROR    Failed to set the param min_p = None, error: 'xllamacpp.xllamacpp.CommonParams' object has no attribute 'min_p'
#启动时候出现上述提示，请问参数该如何传入呢？
我使用的qwen2.5作为测试，启动指令是这样的：xinference launch --model-engine llama.cpp --model-name qwen2.5-instruct --size-in-billions 0_5 --model-format ggufv2 --quantization q2_k --top_k 40 --samplers "top_k;top_p;min_p;temperature;dry;typ_p;xtc"

**Comment by bleakie** [comment]
model.chat(
    prompt,
    chat_history=chat_history,
    generate_config={"max_tokens": 1024, “ top_p”：0.9}
)
#请问xinf client是这样设置吗？

**Comment by bleakie** [comment]
[看到相关文章，不修改sampler](https://blog.csdn.net/python12345_/article/details/146198741) 顺序，依然会出现重复回复的现象

**Comment by bleakie** [comment]
有没有考虑支持下unsloth的qwq，原生版本的这个重复回答的bug太崩溃了

**Comment by bleakie** [comment]
n_batch=10，测试的时候发现并不能并发，只有1个并发，请问参数该如何设置呢？

**Comment by bleakie** [comment]
<img width="679" alt="Image" src="https://github.com/user-attachments/assets/f498b099-63f1-4594-889f-fe33aeb9231e" />

 ERROR    Failed to set the param n_slots = 10, error: 'xllamacpp.xllamacpp.CommonParams' object has no attribute 'n_slots'

**Comment by bleakie** [comment]
请问：您完整的参数输入是什么样子的呢？

**Comment by bleakie** [comment]
我开了2个窗口同时请求，就是上面的情况。同时我也才用异步去并发，也不行

**Comment by bleakie** [comment]
`USE_XLLAMACPP=1 xinference-local --host 0.0.0.0
xinference launch --model-engine llama.cpp --model-name QwQ-32B --size-in-billions 32 --model-format ggufv2 --quantization Q5_K_M --n_ctx 2046 --n_parallel 1 --n_batch 100


srv  params_from_: Chat format: Content-only
slot launch_slot_: id  0 | task 92396 | processing task
slot update_slots: id  0 | task 92396 | new prompt, n_ctx_slot = 12000, n_keep = 0, n_prompt_tokens = 27
slot update_slots: id  0 | task 92396 | kv cache rm [8, end)
slot update_slots: id  0 | task 92396 | prompt processing progress, n_past = 27, n_tokens = 19, progress = 0.703704
slot update_slots: id  0 | task 92396 | prompt done, n_past = 27, n_tokens = 19
srv  params_from_: Chat format: Content-only
slot      release: id  0 | task 92396 | stop processing: n_past = 722, truncated = 0
slot print_timing: id  0 | task 92396 | 
prompt eval time =     104.31 ms /    19 tokens (    5.49 ms per token,   182.15 tokens per second)
       eval time =   13315.54 ms /   696 tokens (   19.13 ms per token,    52.27 tokens per second)
      total time =   13419.85 ms /   715 tokens
slot launch_slot_: id  0 | task 92462 | processing task
slot update_slots: id  0 | task 92462 | new prompt, n_ctx_slot = 12000, n_keep = 0, n_prompt_tokens = 27
slot update_slots: id  0 | task 92462 | need to evaluate at least 1 token to generate logits, n_past = 27, n_prompt_tokens = 27
slot update_slots: id  0 | task 92462 | kv cache rm [26, end)
slot update_slots: id  0 | task 92462 | prompt processing progress, n_past = 27, n_tokens = 1, progress = 0.037037
slot update_slots: id  0 | task 92462 | prompt done, n_past = 27, n_tokens = 1
slot      release: id  0 | task 92462 | stop processing: n_past = 472, truncated = 0
slot print_timing: id  0 | task 92462 | 
prompt eval time =      18.83 ms /     1 tokens (   18.83 ms per token,    53.12 tokens per second)
       eval time =    8488.56 ms /   446 tokens (   19.03 ms per token,    52.54 tokens per second)
      total time =    8507.38 ms /   447 tokens
srv  update_slots: all slots are idle

`
这分别是我启动指令和对应的log，我的请求是并发5个线程，但是依然只有一个并发

**Comment by bleakie** [comment]
n_parallel=n后就可以多路并发，但是会出现重复回答

**Comment by bleakie** [comment]
我是否可以使用key = ctx_shift, value = false，就可以关掉 ctx_shift。

**Comment by bleakie** [comment]
使用xinf-v1.15.1.post

1.我尝试把context shift关掉，使用原模型16000上下文，2路并发，会出现模型异常中断（图1）

<img width="415" alt="Image" src="https://github.com/user-attachments/assets/952f712f-3e48-47ca-9a5c-42abc7018679" />

2.我尝试把context shift关掉，使用原模型32k上下文，2路并发，会出现模型异常中断（图2）

<img width="416" alt="Image" src="https://github.com/user-attachments/assets/1c0c27fc-8306-45e8-991b-c4f345769339" />

**Comment by qinxuye** [comment]
那就是超出上下文了吧。

**Comment by bleakie** [comment]
上下文长度是8000多，而且是跑到一半崩溃（ds-qwen-32）没遇到这个问题

**Comment by bleakie** [comment]
qwen3也会遇到重复回答的情况，n_parallel=1不会出现这个问题。希望早日解决，but ollama就没出现这个情况

**Comment by qinxuye** [comment]
n_parallel > 1 的时候，关闭 context shift 能解决问题吗？

**Comment by bleakie** [comment]
不可以，会出现之前提过的这个错误（然而ctx并没有超）

> 使用xinf-v1.15.1.post
> 
> 1.我尝试把context shift关掉，使用原模型16000上下文，2路并发，会出现模型异常中断（图1）
> 
> <img alt="Image" width="415" src="https://private-user-images.githubusercontent.com/23338843/436385777-952f712f-3e48-47ca-9a5c-42abc7018679.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDY1MDUxMjIsIm5iZiI6MTc0NjUwNDgyMiwicGF0aCI6Ii8yMzMzODg0My80MzYzODU3NzctOTUyZjcxMmYtM2U0OC00N2NhLTlhNWMtNDJhYmM3MDE4Njc5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTA2VDA0MTM0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFkYzM4ZWIwYTNmMTNjNzdhNGZhMWExZWNhZGE2OWVjN2Q0M2NlYzM4MWY3YmQ1YTAyNGNhYWI2NWQzNWExZTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.RkecuPjIV3rUmh8udKb96U42azPQA4RNsEsMVTe5gZo">
> 2.我尝试把context shift关掉，使用原模型32k上下文，2路并发，会出现模型异常中断（图2）
> 
> <img alt="Image" width="416" src="https://private-user-images.githubusercontent.com/23338843/436386140-1c0c27fc-8306-45e8-991b-c4f345769339.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDY1MDUxMjIsIm5iZiI6MTc0NjUwNDgyMiwicGF0aCI6Ii8yMzMzODg0My80MzYzODYxNDAtMWMwYzI3ZmMtODMwNi00NWU4LTk5MWItYzRmMzQ1NzY5MzM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTA2VDA0MTM0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZmM2YwZGQ3NzUyMGJkYTNmMWM4MmE4MThiNmVkMDZiZTBiMzcyMjdjY2ZmNzNmYjIxZGY2MzQ0MWE2NTkwYTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Z6U8Q1pEw9Qu4TNiGf9gjhy13pnr3wHVb5MDXNkFfss">

**Comment by qinxuye** [comment]
看报错倒是抄了，n_ctx 能达到 16000 吗？

**Comment by bleakie** [comment]
达不到，差不多8000

> 看报错倒是超了，n_ctx 能达到 16000 吗？

**Comment by qinxuye** [comment]
得看下启动日志里，真实的 n_ctx 是多少。你配置的是 16000 吗

**Comment by bleakie** [comment]
n_ctx=16000,n_parallel=2

 ('Got error in chat stream: %s', {'code': 500, 'message': 'context shift is disabled', 'type': 'server_error'})


`llama_context: constructing llama_context
llama_context: n_seq_max     = 2
llama_context: n_ctx         = 16000
llama_context: n_ctx_per_seq = 8000
llama_context: n_batch       = 2048
llama_context: n_ubatch      = 512
llama_context: causal_attn   = 1
llama_context: flash_attn    = 0
llama_context: freq_base     = 1000000.0
llama_context: freq_scale    = 1
`

**Comment by qinxuye** [comment]
感觉放不下这么长的。

你可以尝试调整页面上的 n-gpu-layers，卸载部分到内存。

**Comment by qinxuye** [comment]
尺寸一样吗？

**Comment by bleakie** [comment]
一样，使用的都是32b-q5_k_m（其他尺寸也尝试过，ds都是ok的，qwen都不行），且尝试调整n-gpu-layers也不可以

**Comment by bleakie** [comment]
`xinference launch --model-name qwen3 --model-type LLM --model-engine llama.cpp --model-format ggufv2 --size-in-billions 14 --quantization Q5_K_M --n-gpu 2 --replica 1 --n-worker 1 --model-path /mnt/media/sai/data/llm_models/xinference/cache/Qwen3-14B-GGUF/Qwen3-14B-Q5_K_M.gguf --reasoning-content false --n_ctx 14000 --n_parallel 2 --ctx_shift false`

输出到一半的时候会出现：
('Got error in chat stream: %s', {'code': 500, 'message': 'context shift is disabled', 'type': 'server_error'})

**Comment by bleakie** [comment]
> <img alt="Image" width="679" src="https://private-user-images.githubusercontent.com/23338843/434572681-f498b099-63f1-4594-889f-fe33aeb9231e.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDY3ODA5NzAsIm5iZiI6MTc0Njc4MDY3MCwicGF0aCI6Ii8yMzMzODg0My80MzQ1NzI2ODEtZjQ5OGIwOTktNjNmMS00NTk0LTg4OWYtZmUzM2FlYjkyMzFlLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTA5VDA4NTExMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRmODZkZTA0NmI2OTFkZjQ4MmUzODRkZjY2ZjhiOGY0YjU0NGEzYjEzNDg1MzY5MjFjMzZjMGM2OGI4ZTNhZGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1yUqcTQ8mF1nforKJhigsWPT6JY194Yep_rzfAv4PcM">
> ERROR Failed to set the param n_slots = 10, error: 'xllamacpp.xllamacpp.CommonParams' object has no attribute 'n_slots'
> 
> log如上图，设置n_batch=10,n_slots = 10,n_parallel=10，显示以上错误，测试的时候依然不能并发，只有1个任务完成后才会进行下一个。
> 
> ![Image](https://github.com/user-attachments/assets/bc715214-74aa-44ee-9b46-1cbf06ba6744)

参见上图，头大，请问如何解决？
